lib/va_healthcare_messaging/api/messages.rb

```
@@ -75,12 +81,12 @@ def post_create_message_draft(subject:, recipient_id:, category:, body:, id: nil
       ## post_create_message_reply
       ## Replies to a message with the given id, or updates an existing reply, without attachments
       #################################################################################################################
-      def post_create_message_reply(id, subject: nil, body:, draft_id: nil)
-        params = { body: body }
-        params[:subject] = subject unless subject.nil?
-        params[:id] = draft_id unless draft_id.nil?
+      def post_create_message_reply(args = {})
+        params = { body: args[:body] }
+        params[:subject] = args[:subject] unless args[:subject].nil?
+        params[:draft_id] = args[:draft_id] unless args[:draft_id].nil?
 
-        json = perform(:post, "message/#{id}/reply", params.to_json, token_headers)
+        json = perform(:post, "message/#{args[:id]}/reply", params.to_json, token_headers)
```

I think you can get rid of the top 3 lines of this method and just use:

args[:id] and args.only(:subject, :body, :draft_id) no? Not sure why you need to introduce params, you can just work with the existing local variable `args`

alternately, maybe do something like: params = args.reverse_merge(subject: nil, draft_id: nil, body: nil)

or maybe raise an exception if some arguments are required. But reading this as it is, it's not entirely clear why this should be necessary.